src/spikeinterface/sorters/basesorter.py

```diff
@@ -138,9 +138,23 @@ def initialize_folder(cls, recording, output_folder, verbose, remove_existing_fo
 
         rec_file = output_folder / "spikeinterface_recording.json"
         if recording.check_serializablility("json"):
-            recording.dump(rec_file, relative_to=output_folder)
+            # because Windows has different drives we can't do `relative_to`
+            # unless user is on same drive, which is not guaranteed
+            if platform.system() == "Windows":
+                recording.dump(
```

Comment:
You are absically throwing the `relative_to` out in windows.

Wouldn't it better to instead tell users that this option is not supported in windows so they don't run away with the impression that they did indeed save their files with relative paths?

Reply:
So add a warning? Or an assert level and make them re-enter without relative_to?

Reply:
I'm still reading up on this part of the code. I just implemented Sam's suggestion, but haven't read too deep into the rest of the function. Once I made his change I tested and it worked, but I agree Windows users should know. For instance if I do c-> c I could do relative_to without problem. So maybe there is a more universal change for Windows that would be better.

Reply:
Is that a folder that you are saving or a file?
Are you able to load it again?

Reply:
Both. So I have tested this with `run_sorter` and `run_sorter_by_property` with Kilosort2 and Kilosort3 (I am able to read the folder of files with `read_kilosort` and with Mountainsort5 (saves one `.npz` file, which I am able to read with `read_npz_sorting`). I haven't tested any sort of portability yet though. But the provenance files/log files all give absolute paths, so I'm assuming they are not portable, but I can test Monday.

Reply:
Yes, they are not portable. Can you load them with the `base_folder` keyword, this is what I mean,

Reply:
Could you give me a code snippet of what you want me to try please? I have never loaded anything with the `base_folder` keyword.

Reply:
So testing
`recording.dump_to_pickle(..)`
`recording.dump(..)`

both failed with this stack trace
```
  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\base.py:600 in dump
    self.dump_to_json(file_path, relative_to=relative_to, folder_metadata=folder_metadata)

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\base.py:633 in dump_to_json
    dump_dict = self.to_dict(

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\base.py:437 in to_dict
    dump_dict = _make_paths_relative(dump_dict, relative_to)

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\base.py:1014 in _make_paths_relative
    return recursive_path_modifier(d, func, target="path", copy=True)

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\core_tools.py:824 in recursive_path_modifier
    recursive_path_modifier(kwargs, func, copy=False)

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\core_tools.py:847 in recursive_path_modifier
    dc[k] = func(v)

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\base.py:1013 in <lambda>
    func = lambda p: os.path.relpath(str(p), start=relative)

  File ~\anaconda3\envs\spykes\Lib\ntpath.py:754 in relpath
    raise ValueError("path is on mount %r, start on mount %r" % (

ValueError: path is on mount 'C:', start on mount 'D:'
```
trying `sorting.dump_to_pickle` failed with:
```

  File ~\Documents\GitHub\spikeinterface\src\spikeinterface\core\base.py:659 in dump_to_pickle
    file_path.write_bytes(pickle.dumps(dump_dict))

PicklingError: Can't pickle <function BaseExtractor.from_dict at 0x000001C297972020>: it's not the same object as spikeinterface.core.base.BaseExtractor.from_dict
```

So it seems `relative_to` fails on Windows when moving between drives as a feature. Is there something else I can test? Or do you think this would need to be a deeper change then just at the sorting level in this case?